content/articles/index.md

```diff
@@ -0,0 +1,71 @@
+
+
+
+### History of Ray Tracing
+It was created by Robert Goldstein and Arthur Appel in the 1960s. According to [Wikipedia](https://en.wikipedia.org/wiki/Ray_tracing_(graphics)#Algorithm_overview), in 1968, Arthur Appel was the first to use a computer to generate shaded images using ray tracing. He used ray tracing to figure out which surface was closest to the camera at each image point, then traced secondary rays to the source of light from each shaded point to see if it was in shadow or not.
+
+Later, by simulating the photographic process in reverse, Robert Goldstein used ray tracing is to create shaded images of solids. He used a non-recursive ray tracing based algorithm known as ray casting. Ray tracing is widely regarded as the most effective method of image synthesis.
+
+
+### Types of Ray Tracing Algorithm
+Various methods are induced to make ray tracing effective. There are three types of ray tracing algorithms, they include; hybrid ray tracing algorithm, forward ray tracing algorithm, backward ray tracing algorithm.
+
+
+#### 1. Hybrid Ray Tracing Algorithm
+It's a combination of backward and forward ray tracing techniques hence making it more effective since it has the benefits of both techniques.   
+
+A first pass through the scene is rendered using a basic raster image processing system such as [Vulkan](https://vulkan-tutorial.com/). The first pass is then followed by a ray tracing phase which adds more information to the image created in the first pass.
+
+
+#### 2. Forward Ray Tracing Algorithm
+It can also be referred to as photon/light ray tracing. This is because it follows photons from the source of light to the specific object. Forward ray tracing is the best technique for determining the color of the object despite it being the most inefficient method due to the fact that several rays from the source of light never make it through the view plane and into the eye.
+
+In Forward ray tracing it's assumed that the reflected ray intersects the eye's surface even though in reality rays are reflected in every direction. Light from the source will travel in a straight direction until it hits a surface where it then changes to other likely directions. 
+
+
+#### 3. Backward Ray Tracing Algorithm
+In backward ray tracing, an eye ray is produced that travels through the view plane and into space. The eye should be in contact with the object for it to be hit by the eye ray hence being from the point of view plane. The ray tracer then finds out the exact coloring and shade of this point in the view plane after the light beam bounces around. 
+
+Like forward ray tracing, backward ray tracing also has an assumption. It's assumed that only light rays that are traveling through the view plane and then into the eye to produce the final image. The assumption may not be true since many light rays may travel through the view plane in some instances. Backward ray tracing is also known as eye-tracing.
+

```

Comment:
Before the conclusion, provide a section for the applications of ray tracing algorithm.